### PR TITLE
ROX-13480: Add view active yamls tab

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR is an iterative addition to the network policy simulator. We will display the active network policy yamls in the "view active yamls" tab

Screenshot 2022-12-09 at 12 22 15 PM

## Checklist
- [ ] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

### CODE REVIEW

File: ui/apps/platform/src/Containers/NetworkGraph/components/NetworkBreadcrumbs.tsx [hunk 1]
```diff
@@ -14,10 +14,7 @@ type NetworkBreadcrumbsProps = {
 function NetworkBreadcrumbs({ clusters = [] }: NetworkBreadcrumbsProps) {
     const { searchFilter, setSearchFilter } = useURLSearch();
 
-    const { cluster: selectedClusterName } = getScopeHierarchy(searchFilter);
-
-    const selectedClusterId =
-        clusters.find((cluster) => cluster.name === selectedClusterName)?.id || '';
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/components/NetworkBreadcrumbs.tsx:20]: I got extensive changes to this section in my currently open PR, to make it cleaner when adding support for deployments. In fact, it passes a minimum cluster object, just name and ID, after that change.

https://github.com/stackrox/stackrox/pull/4093/files#diff-3703b64f72da48d77b30d34395ab1993f2fd70e6d6c20e5585a999834ca19b2d

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ function NetworkGraphPage() {
     const [simulationQueryValue] = useURLParameter('simulation', undefined);
 
     const {
-        cluster: clusterFromUrl,
+        cluster: clusterIdFromUrl,
```
[the remark this answers is not in the document]
Comment (sachaudh) [reply ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx:63]: Sure we can do that. I saw you did that with your latest PR so I'll rebase and use that

File: ui/apps/platform/src/Containers/NetworkGraph/simulation/NetworkPolicySimulatorSidePanel.tsx [hunk 3]
```diff
@@ -14,13 +14,20 @@ import {
 } from '@patternfly/react-core';
 
 import useTabs from 'hooks/patternfly/useTabs';
+import ViewActiveYamls from './ViewActiveYAMLs';
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/NetworkGraph/simulation/NetworkPolicySimulatorSidePanel.tsx:17]: nit: keep casing of file and component consistent

It doesn't really matter whether they are both `ViewActiveYamls` or `ViewActiveYAMLS`.

**Review by vjwilson — APPROVED** [review]
Just suggestions for later

File: ui/apps/platform/src/Containers/NetworkGraph/hooks/apis/useFetchActiveYAMLs.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+import { fetchNetworkPoliciesByClusterId } from 'services/NetworkService';
+import { NetworkPolicy } from 'types/networkPolicy.proto';
+
+type Result = { isLoading: boolean; networkPolicies: NetworkPolicy[]; error: string | null };
+
+const defaultResultState = { networkPolicies: [], error: null, isLoading: true };
```
[the remark this answers is not in the document]
Comment (sachaudh) [reply ui/apps/platform/src/Containers/NetworkGraph/hooks/apis/useFetchActiveYAMLs.ts:7]: Is there a specific one that comes to mind. I think this is pretty much how they all are. I think the only difference is that I don't let the `networkPolicies` be null. I can change that tho

**Pull request merged**